libcxx/test/support/nasty_string.h

```diff
@@ -40,8 +40,8 @@
 #ifndef TEST_HAS_NO_NASTY_STRING
 // Make sure the char-like operations in strings do not depend on the char-like type.
 struct nasty_char {
-  template <typename T>
-  friend auto operator<=>(T, T) = delete;
+  template <typename T, typename U>
+  friend auto operator<=>(T, U) = delete;
```

Comment:
Some tests fail with the unstable ABI (See #107747). This is the easiest remedy to make `std::__wrap_iter<nasty_string*>` a Cpp17RandomAccessIterator (otherwise this overload is found for `it <= it`) which some assertions rely on 

It would make sense as a separate patch too, since these tests currently fail with unstable ABI + hardening right now.